### ISSUE TITLE
Get FF compiling on more recent Visual Studio

### ISSUE THREAD

**Issue by AfterShockFF** [fortressforever/fortressforever#221]
Currently we can only develop on Visual Studio 2005 which is very old. From a quick test it seems like this is something to do with tier0 / tier1 valve libraries.

Upgrading would mean we can take advantage of a load of new features and stuff like Resharper C++.

**Comment by DexterHaslem** [comment]
I've also been wanting to do this, and have tried in the past. another issue was luabind not building. newer version of luabind worked, but needed newer version of boost and I got stuck somewhere else I can't recall

**Comment by DexterHaslem** [comment]
oh yea, only tier1 can be build from sources (in everything project), so screwed on tier0,  2,3 being built  on older versions (lib\public & lib-vc7\public) as far as i know. I'm not super familiar with the ABI changes over the yhears

a licensee could try to rebuild them but trying to build on newer version of VC++ would probably cause a million problems

**Comment by AfterShockFF** [comment]
So it looks like we can't link to the old tier0/2/3 lib files with the newer compilers because they were compiled in ancient C++ and the compiler has had breaking changes since then, so we would either need to:
- Get hold of the source code and then update the source code to deal with the minor little things that have changed, then recompile in the newest compiler
- Keep compiling in VS2005 but somehow work out how to work in a newer VS and backport the project changes in order to compile (some automated project-change detection script?)
- Work out how to wire up a newer VS to use the old compiler, but still take advantage of all the newest VS features

**Comment by AfterShockFF** [comment]
It looks like the 3rd option is probably possible - wire new visual studio up to use the old compiler. It's a bit of a pain to do, and I don't have it fully working yet, but you can use something called Daffodil:

https://daffodil.codeplex.com/

You need to install VS2010 then the latest VS afterwards.

http://stackoverflow.com/questions/15029762/is-it-possible-to-use-the-vc-6-compiler-in-visual-studio-2012/16487010

**Comment by AfterShockFF** [comment]
Ok I can't get this to work - I'm sure someone else probably could if anyone wants to give it a try. The trick is to get it working in 2010 first, it seems. I got as far as making 2012 target the old 2005 compiler but it throws an error:

`C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\Platforms\Win32\Microsoft.Cpp.Win32.Targets(153,5): error MSB6006: "CL.exe" exited with code -1073741515.`

Which I can't figure out, there doesn't seem to be any useful feedback on exactly what's going wrong.

**Comment by DexterHaslem** [comment]
I havent looked at all but does this daffodil let you specify which platform SDK the compiler tries to use? I had vs2005 blow up with no error in past trying to use latest platform sdk

**Comment by AfterShockFF** [comment]
Yeah It does. I've actually almost got it working. I get it compiling just the linker blows up. It says it cant find odbc32 which shouldnt be needed. If i install the platform SDK data tools then it gets past that error, then it says something about libcpmtd.lib, which again shouldnt be necessary. I think its something to do with the associated dependencies or perhaps pulling in the wrong folders.

If you want to give it a bash follow the instructions on that stackoverflow question, install 2010 and then daffodil then you can select v80 from the platformToolset dropdown. You'll need to manually add the VS and VC directories to the project settings (similar to the instructions on compiling FF on the front page of github) and the platform SDK dirs also - because these extra directories are stored in the actual project files in 2010+ (i think for 2005 they were stored on a per installation thing).
